docs/enhancements/metricsConfig_redesign.md

```diff
@@ -0,0 +1,119 @@
+Adding labels only refers to the labels propagated from the Cluster Deployment, any other strings as labels cannot be configured this way.
+
+
+## Proposal
+Allow HiveConfig.Spec.metricsConfig to look like
+```
+HiveConfig:
+  spec:
+    metricsConfig:
+      clusterDeploymentRelatedMetrics:
+      - name: hive_foo_counter
+        minimumDuration: 10m
+        clusterDeploymentLabelSelector:
+          matchLabels:
+            hive.openshift.io/aro-snowflake: "true"
+          matchExpressions:
+            - key: hive.openshift.io/limited-support
+              operator: NotIn
+              values: 
+                - "true"
+            - key: hive.openshift.io/limited-support
+              operator: NotExists
+        additionalClusterDeploymentLabels:
+          prom_label_name: hive.openshift.io/cd-label-key
+```
+### clusterDeploymentRelatedMetrics
+We are only considering with metrics that are calculated whenever we parse the cluster deployment, or is tied to a cluster deployment, so we will group them all under hiveConfig.spec.metricsConfig.clusterDeploymentRelatedMetrics. Each metric that needs to be customized would need a separate entry.
+Implementation would require a separate class that would encapsulate all the clusterDeploymentRelatedMetrics, and related methods for registering and observing the metrics.
+
+#### minimumDuration
+Deprecate the current implementation of hiveConfig.spec.metricsConfig.metricsWithDuration, and change it to be reported as metricsConfig.clusterDeploymentRelatedMetrics[$name].minimumDuration. The implementation of using the duration as a threshold before we report the metric stays the same.
+
+#### additionalClusterDeploymentLabels
+Deprecate current implementation of hiveConfig.spec.metricsConfig.additionalClusterDeploymentLabels and change it to be reported as metricsConfig.clusterDeploymentRelatedMetrics[$name].additionalClusterDeploymentLabels. It's implementation will not change.
+
+#### clusterDeploymentLabelSelector
+This would be a new feature, of type [LabelSelector](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#LabelSelector), and we'd use the LabelSelector.MatchLabels and/or LabelSelector.MatchExpressions to match the conditions in order to decide if a metric should be reported.
+This encapsulates slightly more advanced filter logic over the existing clusterdeployment labels. In the example above, hive_foo_counter metric will only be reported for the cluster deployments that are labelled with aro-snowflake, not in limited support, and has the metric value > 10m.
+
+### Implementation Details / Notes
+- In order to implement this change, we will have to deprecate the existing hiveconfig.spec.metricsConfig.metricsWithDuration and hiveconfig.spec.metricsConfig.additionalClusterDeploymentLabels.`
```

Comment:
Would you mind elaborating a little bit on the plan for deprecation? Are we planning to support both mechanisms for $time? Will they be mutually exclusive for that period, resulting in a hard error (operator crashes) if both are configured? Etc.